projects/go-lib/src/lib/components/go-side-nav/go-side-nav/go-side-nav.component.spec.ts

```diff
@@ -0,0 +1,122 @@
+import { NavGroup } from '../nav-group.model';
+import { Component } from '@angular/core';
+
+@Component({
+  template: ''
+})
+export class FakeComponent {
+}
+
+describe('GoSideNavComponent', () => {
+  let component: GoSideNavComponent;
+  let fixture: ComponentFixture<GoSideNavComponent>;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      declarations: [
+        FakeComponent,
+        GoNavGroupComponent,
+        GoNavItemComponent,
+        GoSideNavComponent
+      ],
+      imports: [
+        GoIconModule,
+        RouterTestingModule
+      ],
+      providers: [
+        GoConfigService
+      ]
+    })
+    .compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(GoSideNavComponent);
+    component = fixture.componentInstance;
+
+    component.menuItems = [
+      {
+        routeTitle: 'Group 1',
+        subRoutes: [
+          {
+            routeTitle: 'Item1',
+            route: '/'
+          }
+        ]
+      },
+      {
+        routeTitle: 'Group 2',
+        subRoutes: [
+          {
+            routeTitle: 'Item2',
+            route: '/item_2'
+          },
+          {
+            routeTitle: 'SubGroup',
+            subRoutes: [
+              {
+                routeTitle: 'SubGroupItem',
+                route: '/sub_group_item'
+              }
+            ]
+          }
+        ]
+      }
+    ];
+
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  describe('ngOnInit', () => {
+    it('should expand navGroup of route', () => {
+      component.ngOnInit();
+      expect(component.menuItems[0]['expanded']).toBe(true);
+    });
+  });
+
+  describe('closeNavs', () => {
+    beforeEach(() => {
+      component.menuItems[1]['expanded'] = true;
+      component.menuItems[0]['expanded'] = false;
+    });
+
+    it('sets all navGroups to closed beside the one passed in', () => {
+      component.closeNavs(component.menuItems[0] as NavGroup);
+
+      expect(component.menuItems[1]['expanded']).toBe(false);
+      expect(component.menuItems[0]['expanded']).toBe(true);
+    });
+
+    it('should keep top level open if nested group expanded', () => {
+      const nestedGroup: NavGroup = component.menuItems[1]['subRoutes'][1];
+
+      component.closeNavs(nestedGroup);
+
+      expect(component.menuItems[1]['expanded']).toBe(true);
+      expect(nestedGroup.expanded).toBe(true);
+    });
+
+    it('sets nested navGroups all closed if they\'re not the one passed in', () => {
+      const nestedGroup: NavGroup = component.menuItems[1]['subRoutes'][1];
+      nestedGroup.expanded = true;
+
+      component.closeNavs(component.menuItems[0] as NavGroup);
+
+      expect(nestedGroup.expanded).toBe(false);
+      expect(component.menuItems[1]['expanded']).toBe(false);
+      expect(component.menuItems[0]['expanded']).toBe(true);
+    });
+  });
+});
```

Comment:
Should we test the other methods in this file as well? I know they're private, but we can test them still.

Reply:
So here's my thoughts on this. I think as long as we test everything in the public methods that we actually want to happen, there's not really a reason to test the private methods. We don't really care how the menuItems gets expanded, we just care that when we click on an item it gets set to expanded. Private methods are basically just helpers and ways to not duplicate code. If we write good tests around the public methods that test everything and then also test the private methods it creates duplicate tests. I think this response on stack overflow has really good insight on testing private methods (https://stackoverflow.com/questions/105007/should-i-test-private-methods-or-only-public-ones/47401015#47401015). Thoughts?

Reply:
@StevenUlmer yeah that's fine. I forgot we already talked and decided on this. Disregard my comments on them.